src/test/java/org/pesmypetcare/webservice/service/MedicationServiceTest.java

```diff
@@ -0,0 +1,152 @@
+        verify(MedicationDao).createMedication(isA(String.class), isA(String.class), isA(String.class), isA(MedicationEntity.class));
+    }
+
+    @Test
+    public void shouldReturnNothingWhenMedicationDeleted() {
+        service.deleteByDateAndName(owner, petName, dateName);
+        verify(MedicationDao).deleteByDateAndName(isA(String.class), isA(String.class), isA(String.class));
+    }
+
+    @Test
+    public void shouldReturnNothingWhenAllMedicationsDeleted() throws DatabaseAccessException {
+        service.deleteAllMedications(owner, petName);
+        verify(MedicationDao).deleteAllMedications(isA(String.class), isA(String.class));
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenAllMedicationDeleteFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(MedicationDao).deleteAllMedications(any(String.class), isA(String.class));
+            service.deleteAllMedications(owner, petName);
+        }, "Should return a database access exception when a Medication deletion fails");
+    }
+
+    @Test
+    public void shouldReturnMedicationEntityWhenMedicationRetrieved() throws DatabaseAccessException {
+        when(MedicationDao.getMedicationData(owner, petName, dateName)).thenReturn(MedicationEntity);
+        MedicationEntity Medication = service.getMedicationData(owner, petName, dateName);
+        assertSame(MedicationEntity, Medication, "Should return a Medication entity");
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenGetMedicationRequestFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(MedicationDao).getMedicationData(any(String.class), any(String.class),
+                    isA(String.class));
+            service.getMedicationData(owner, petName, dateName);
+        }, "Should return an exception when retrieving a Medication fails");
+    }
+
+    @Test
+    public void shouldReturnMedicationEntityListWhenGetSetOfMedicationsRetrieved() throws DatabaseAccessException {
+        when(MedicationDao.getAllMedicationData(owner, petName)).thenReturn(MedicationList);
+        List<Map<List<String>, Object>> list = service.getAllMedicationData(owner, petName);
+        assertSame(MedicationList, list, "Should return a list of Medication entities");
+    }
+
+    @Test
+    public void shouldReturnDatabaseAccessExceptionWhenGetSetOfMedicationsRequestFails() {
+        assertThrows(DatabaseAccessException.class, () -> {
+            doThrow(DatabaseAccessException.class).when(MedicationDao).getAllMedicationData(any(String.class), isA(String.class));
+            service.getAllMedicationData(owner, petName);
+        }, "Should return an exception when retrieving a set of Medications fails");
+    }
+
+    @Test
+    public void shouldReturnMedicationEntityListWhenGetMedicationsBetweenDatesRetrieved() throws DatabaseAccessException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 122).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)